packages/core/src/build-utils/build/utils/loader.ts

```diff
@@ -190,7 +241,10 @@ export function createLoaderContextTrap(
               if (options.hasOptions) {
                 return omit(target.query, [Loader.LoaderInternalPropertyName]);
               }
-              return target.resourceQuery;
+              const postfixRE = /[#?].*$/s;
+              const resource = options?.loader;
+              const loaderQuery = (postfixRE.exec(resource) ?? '').toString();
```

Comment:
We are unable to handle situations where folders contain a question mark (?). We need to confirm whether webpack and rspack can handle this.

Reply:
https://github.com/vitejs/vite/issues/15726

Reply:
webpack https://github.com/webpack/webpack/blob/2953d23a87d89b3bd07cf73336ee34731196c62e/lib/util/identifier.js#L311
rspack https://github.com/web-infra-dev/rspack/blob/d22f049d4bce4f8ac20c1cbabeab3706eddaecc1/packages/rspack/src/loader-runner/index.ts#L47

Neither "webpack" nor "vite" nor "rspack" handle situations where folders contain a "?"